docs/extend/modular-resources/upload/_index.md

```diff
@@ -0,0 +1,234 @@
+     </tr>
+     <tr>
+       <td><code>description</code></td>
+       <td>string</td>
+       <td><strong>Required</strong></td>
+       <td>A description of your module and what it provides.</td>
+     </tr>
+     <tr>
+       <td><code>models</code></td>
+       <td>object</td>
+       <td><strong>Required</strong></td>
+       <td>A list of one or more <a href="/extend/modular-resources/key-concepts/#models">models</a> provided by your custom module. You must provide at least one model, which consists of an <code>api</code> and <code>model</code> key pair.</td>
+     </tr>
+     <tr>
+       <td><code>entrypoint</code></td>
+       <td>string</td>
+       <td><strong>Required</strong></td>
+       <td>The name of the file that starts your module program. This can be a compiled executable, a script, or an invocation of another program.</td>
+     </tr>
+   </table>
+
+   For example, the following represents the configuration of an example `my-module` module in the `acme` namespace:
+
+   ```json {class="line-numbers linkable-line-numbers"}
+   {
+     "name": "acme:my-module",
+     "visibility": "public",
+     "url": "https://github.com/acme-co-example/my-module",
+     "description": "An example custom module.",
+     "models": [
+       {
+         "api": "rdk:component:generic",
+         "model": "acme:demo:my-model"
+       }
+     ],
+     "entrypoint": "my-module.sh"
+   }
+   ```
+
+   {{% alert title="Important" color="note" %}}
+   If you are publishing a public module (`visibility: "public"`), the [namespace of your model](/extend/modular-resources/key-concepts/#namespace-1) must match the [namespace of your organization](/manage/fleet/organizations/#create-a-namespace-for-your-organization).
+   In the example above, the model namespace is set to `acme` to match the owning organization's namespace.
+   If the two namespaces do not match, the command will return an error.
+   {{% /alert %}}
+
+   See [The `meta.json` file](/manage/cli/#the-metajson-file) for more information.
+
+1. Run `viam module update` to register the configuration changes you just made to `meta.json` with the Viam Registry.
+   Run this command from within the same directory as your `meta.json` file:
+
+      ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+      viam module update
+      ```
+
+   On a successful update, the command will return a link to the updated module in the Viam Registry.
+
+1. Package your custom module to get it ready to upload to the Viam Registry.
+   Currently, the Registry only supports `tar.gz` or `tar.xz` format.
+   Use the command below specific for the language of your module:
+
+   - To package a module written in Go, run the following command from the same directory as your `meta.json` file:
+
+     ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+     tar -czf module.tar.gz module.go
+     ```
```

Comment:
```suggestion
     ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
     tar -czf module.tar.gz <path/to/your/compiled/module>
     ```
```

creating a tar of the `.go` source file will confuse people

Reply:
I have no idea if `<path/to/your/compiled/module>` is the right way to express this though.. something like `bin/module` might be clearer??

Reply:
You might want to have something like

```sh
go build -o bin/module ./module/main.go
tar -cxf module.tar.gz bin/module
```

? I don't know enough about how the docs are set up.

Reply:
Better to be explicit, agreed. Updating.